### PR TITLE
Pad the input on transposed conv forward with output_padding

### PR DESCRIPTION
**Pull request by JackCaoG**
This pr is to solve the issue brought up in https://github.com/pytorch/xla/issues/2333.

On `transposed_conv_forward` we added `output_padding` to the output size. Therefore we also need to pad the input to match the output size and unpad the grad_input(on `transposed_conv_backward`) to make it match the input size.

### CODE REVIEW

File: torch_xla/csrc/convolution.cpp [hunk 1]
```diff
@@ -98,6 +98,34 @@ namespace {
...
+    xla::int64 pad_to_input = ((input_spatial - 1) * stride[spatial_dim] +
+                               output_padding[spatial_dim]) /
+                                  stride[spatial_dim] +
+                              1 - input_spatial;
+    xla::PaddingConfig::PaddingConfigDimension* dims =
+        padding_config.add_dimensions();
+    dims->set_edge_padding_high(unpad ? -1 * pad_to_input : pad_to_input);
+  }
+  xla::XlaOp zero = xla::Zero(input.builder(), input_shape.element_type());
+  return xla::Pad(input, zero, padding_config);
```
Comment (dlibenzi) [inline torch_xla/csrc/convolution.cpp:126]: I'd use the `PadToSize()` API here.
Just added a small optimization.

https://github.com/pytorch/xla/pull/2352

File: test/cpp/test_aten_xla_tensor.cpp [hunk 2]
```diff
@@ -8601,13 +8601,13 @@ TEST_F(AtenXlaTensorTest, TestConv2DBackward) {
...
   int out_channels = 8;
-  int kernel_size = 5;
+  int kernel_size = 4;
   for (int stride = 1; stride <= 2; ++stride) {
     for (int padding = 0; padding <= 1; ++padding) {
       for (int dilation = 1; dilation <= 2; ++dilation) {
         for (int output_padding = 0;
-             output_padding < std::min(stride, dilation); ++output_padding) {
-          for (bool with_bias : {true, false}) {
+             output_padding < std::max(stride, dilation); ++output_padding) {
```
[the remark this answers is not in the document]
Comment (JackCaoG) [reply test/cpp/test_aten_xla_tensor.cpp:8609]: This test has always been passing, it does not trigger the corner cases mentioned in the issue.

`output_padding < std::max(stride, dilation)` is needed, I can undo the rest and check if test failed without my change. 

Pad to input is calculated as 
```
pad_to_input = ((input_spatial - 1) * stride[spatial_dim] +
                               output_padding[spatial_dim]) /
                                  stride[spatial_dim] +
                              1 - input_spatial;
``` 

In the existing test since  pad_to_input will always be 0. Since `output_padding < std::min(stride, dilation)`, output_padding = 1 only when stide = 2 and dilation = 2. In this case pad_to_input will be 0. For the rest of the cases output_padding is 0 so `pad_to_input` will be 0 too.

[the remark this answers is not in the document]
Comment (JackCaoG) [reply test/cpp/test_aten_xla_tensor.cpp:8609]: I think this is the intention anyway, pytorch requires `output_padding` to be smaller than either `stride` or `dilation`.

File: test/cpp/test_aten_xla_tensor.cpp [hunk 3]
```diff
@@ -8693,7 +8693,7 @@ TEST_F(AtenXlaTensorTest, TestTransposedConv3DBackward) {
     for (int padding = 0; padding <= 1; ++padding) {
       for (int dilation = 1; dilation <= 2; ++dilation) {
         for (int output_padding = 0;
-             output_padding < std::min(stride, dilation); ++output_padding) {
+             output_padding < std::max(stride, dilation); ++output_padding) {
```
[the remark this answers is not in the document]
Comment (JackCaoG) [reply test/cpp/test_aten_xla_tensor.cpp:8696]: Same as above test.

File: test/cpp/test_aten_xla_tensor.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (JackCaoG) [reply test/cpp/test_aten_xla_tensor.cpp:8609]: verify that 
without my change and with `output_padding < std::min(stride, dilation); ++output_padding`. -> test passed
without my change and with `output_padding < std::max(stride, dilation); ++output_padding`. -> test failed
with my change and with `output_padding < std::max(stride, dilation); ++output_padding`. -> test passed

same result for the conv3d test

File: torch_xla/csrc/xla_lower_util.cpp [hunk 4]
```diff
@@ -303,7 +303,6 @@ xla::XlaOp PadToSize(xla::XlaOp input, absl::Span<const xla::int64> size,
     auto* dims = padding_config.add_dimensions();
     dims->set_edge_padding_low(0);
     dims->set_interior_padding(0);
-    XLA_CHECK_GE(size[i], input_shape.dimensions(i));
```
Comment (JackCaoG) [inline torch_xla/csrc/xla_lower_util.cpp:306]: There is a valid use case to pad the size < origional_size, removed this check
